hw/bsp/dialog_da1469x-dk-pro/otp_tool.py

```diff
@@ -0,0 +1,538 @@
+    for entry in trim:
+        data.append(int(entry, 16))
+
+    send_otp_config_payload(uart, data)
+
+
+@click.option('-u', '--uart', required=True, help='uart port')
+@click.command(help='Disable development mode')
+def disable_development_mode(uart):
+
+    send_otp_config_payload(uart, [0x70000000])
+
+
+@click.option('-u', '--uart', required=True, help='uart port')
+@click.command(help='Enable secure boot')
+def enable_secure_boot(uart):
+
+    send_otp_config_payload(uart, [0x500000cc, 0x1])
+
+
+@click.option('-u', '--uart', required=True, help='uart port')
+@click.command(help='Write lock OTP QSPI key area')
+def disable_qspi_key_write(uart):
+
+    send_otp_config_payload(uart, [0x500000cc, 0x40])
+
+
+@click.option('-u', '--uart', required=True, help='uart port')
+@click.command(help='Read lock OTP QSPI key area')
+def disable_qspi_key_read(uart):
+
+    send_otp_config_payload(uart, [0x500000cc, 0x80])
+
+
+@click.option('-u', '--uart', required=True, help='uart port')
+@click.command(help='Write lock OTP user key area')
+def disable_user_key_write(uart):
+
+    send_otp_config_payload(uart, [0x500000cc, 0x10])
+
+
+@click.option('-u', '--uart', required=True, help='uart port')
+@click.command(help='Read lock OTP user key area')
+def disable_user_key_read(uart):
+
+    send_otp_config_payload(uart, [0x500000cc, 0x20])
+
+
+@click.option('-u', '--uart', required=True, help='uart port')
+@click.command(help='Write lock OTP signature key area')
+def disable_signature_key_write(uart):
+
+    send_otp_config_payload(uart, [0x500000cc, 0x8])
+
+
+@click.option('-u', '--uart', required=True, help='uart port')
+@click.command(help='Disable CMAC debugger')
+def disable_cmac_debugger(uart):
+
+    send_otp_config_payload(uart, [0x500000cc, 0x4])
+
+
+@click.option('-u', '--uart', required=True, help='uart port')
+@click.command(help='Disable SWD debugger')
+def disable_swd_debugger(uart):
+
+    send_otp_config_payload(uart, [0x500000cc, 0x2])
+
+
+@click.option('-u', '--uart', required=True, help='uart port')
+@click.command(help='Close out OTP configuration script')
+def close_config_script(uart):
+
+    send_otp_config_payload(uart, [0x0])
+
+
+@click.option('-u', '--uart', required=True, help='uart port')
+@click.command(help='Initialize blank OTP Config script')
+def init_config_script(uart):
+    try:
+        ser = serial.Serial(port=uart, baudrate=115200, timeout=1,
+                            bytesize=8, stopbits=serial.STOPBITS_ONE)
+    except serial.SerialException:
+        raise SystemExit("Failed to open serial port")
+
+    # length is unused, so set to 0
+    cmd = cmd_append_value(0xaa55aa55, Cmd.OTP_INIT, 0)
+    msg = struct.pack('III', *cmd)
+
+    try:
+        ser.write(msg)
+    except serial.SerialException:
+        raise SystemExit("Failed to write to %s" % uart)
+
+    data = ser.read(16)
+    if len(data) != 16:
+        raise SystemExit("Failed to receive response, exiting")
```

Comment:
Since this pattern is copied all over, one option would be to add a helper function which reads from the handle and raises the error, then you could just use something like `data = exact_read(handle, size)` and leave the size check + raise inside that helper function.